mlflow/store/model_registry/sqlalchemy_store.py

```diff
@@ -205,6 +205,48 @@ def list_registered_models(self):
             return [sql_registered_model.to_mlflow_entity()
                     for sql_registered_model in session.query(SqlRegisteredModel).all()]
 
+    def search_registered_models(self, filter_string):
+        """
+        Search for model versions in backend that satisfy the filter criteria.
+
+        :param filter_string: A filter string expression. Currently supports a single filter
+                              condition either name of model like ``name = 'model_name'``
+
+        :return: PagedList of :py:class:`mlflow.entities.model_registry.ModelVersion`
+                 objects.
+        """
+        parsed_filter = SearchUtils.parse_filter_for_model_registry(filter_string)
+        if len(parsed_filter) == 0:
+            conditions = []
+        elif len(parsed_filter) == 1:
+            filter_dict = parsed_filter[0]
+            if filter_dict["comparator"] not in \
+                    SearchUtils.VALID_REGISTERED_MODEL_SEARCH_COMPARATORS:
+                raise MlflowException('Model Registry search filter only supports equality(=) '
+                                      'comparator, case-sensitive partial match (LIKE),'
+                                      'and case-insensitive partial match (ILIKE).'
+                                      'Input filter string: %s' % filter_string,
+                                      error_code=INVALID_PARAMETER_VALUE)
+            if filter_dict["key"] == "name":
+                if filter_dict["comparator"] == "LIKE":
+                    conditions = [SqlRegisteredModel.name.like(filter_dict["value"])]
+                elif filter_dict["comparator"] == "ILIKE":
+                    conditions = [SqlRegisteredModel.name.ilike(filter_dict["value"])]
+                else:
+                    conditions = [SqlRegisteredModel.name == filter_dict["value"]]
+            else:
+                raise MlflowException('Invalid filter string: %s' % filter_string,
+                                      error_code=INVALID_PARAMETER_VALUE)
+        else:
+            raise MlflowException('Model Registry expects filter to be like '
```

Comment:
Change message to something like --

> Invalid filter string '%s'. Search registered models supports filter expression of the type "name <op> 'model-name'". Supported <op> are equality (=) comparator...

(you can reuse that part of the message from the above error message. I would create a local variable so you do not need to change two places when we add more operator support)

Reply:
cc: @andreakress